activerecord/test/cases/adapters/postgresql/date_test.rb

```diff
@@ -39,10 +39,4 @@ def test_bc_date_year_zero
     topic = Topic.create!(last_read: date)
     assert_equal date, Topic.find(topic.id).last_read
   end
-
-  def test_date_decoder
-    date = ActiveRecord::Base.connection.select_value("select '2024-01-01'::date")
-    assert_equal Date.new(2024, 01, 01), date
-    assert_equal Date, date.class
-  end
```

Comment:
I think you should keep this test, you just need to turn the config on for the duration of the test. You could also write another test that shows what happens when the config is off.